### PR TITLE
Feat improve aggregations

### LINKED ISSUE #141: Improve aggregations code to be more dynamic
The aggregations code is not as dynamic as it should. 
- [x] The idea is to define the table attributes by the hash that is generated from the data.
- [x] As if we would like to add another aggregated element of data we would not modify the tab at all
- [x] The aggregated data should also be dynamic and defined by some arguments passed to a method instead of doing some manual stuff. This can be achieved by doing a block method that would take some custom code if required

### PR DESCRIPTION
**Pull request by hyperionel**
Fixes #141 and #166

### CODE REVIEW

**Comment by Chocksy** [comment]
@ungureanu-bogdan-sorin don't forget to rebase this.

**Comment by hyperionel** [comment]
rebased

File: app/models/grouped_issue.rb [hunk 1]
```diff
@@ -8,8 +9,18 @@ class GroupedIssue < ActiveRecord::Base
...
+  before_save :generate_uuid
+
+  def generate_uuid
+    if self.uuid.nil?
+      self.uuid = loop do
+        uuid = SecureRandom.hex(4)
+        break uuid unless self.class.exists?(uuid: uuid)
+      end
+    end
+  end
```
Comment (Chocksy) [inline app/models/grouped_issue.rb]: Isn't friendly_id generating a uuid for us? Why are you doing this?

File: db/migrate/20160627082118_add_uuid_to_grouped_issues.rb [hunk 2]
```diff
@@ -0,0 +1,6 @@
+class AddUuidToGroupedIssues < ActiveRecord::Migration
+  def change
+    add_column :grouped_issues, :uuid, :string
+    add_index :grouped_issues, :uuid
```
Comment (Chocksy) [inline db/migrate/20160627082118_add_uuid_to_grouped_issues.rb:4]: We need an unique index on this. If you set just index that is not enough.

File: app/models/grouped_issue.rb (hunk 1 above)
Comment (hyperionel) [inline app/models/grouped_issue.rb]: It generates an unique id based on the column you assign it to. Tried to assign it to the message column but the url ends up looking bad.

File: config/initializers/friendly_id.rb [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+  #
+  # This is significantly more convenient but may not be appropriate for
+  # all applications, so you must explicity opt-in to this behavior. You can
+  # always also configure it on a per-model basis if you prefer.
+  #
+  # Something else to consider is that using the :finders addon boosts
+  # performance because it will avoid Rails-internal code that makes runtime
+  # calls to `Module.extend`.
+  #
+  # config.use :finders
```
Comment (Chocksy) [inline config/initializers/friendly_id.rb]: i think we should use finders here. Uncomment this line. I don't wanna use `friendly.find()` calls in our code

Comment (Chocksy) [inline config/initializers/friendly_id.rb]: I see you have this set in the model but we wanna use it everywhere so you will be able to remove it from there i think.

File: db/migrate/20160707095431_add_slug_to_grouped_issues.rb [hunk 4]
```diff
@@ -0,0 +1,7 @@
+class AddSlugToGroupedIssues < ActiveRecord::Migration
+  def change
+    remove_column :grouped_issues, :uuid, :string
+    add_column :grouped_issues, :slug, :string
+    add_index :grouped_issues, :slug
```
Comment (Chocksy) [inline db/migrate/20160707095431_add_slug_to_grouped_issues.rb]: You forgot to add the uniq index for the slug column.. I'll add it here but you have to run `rake db:schema:load`!!

**Pull request merged**